src/Cache/FileHandler.php

```diff
@@ -60,13 +60,36 @@ public function write(CacheInterface $cache)
     {
         $content = $cache->toJson();
 
+        if (file_exists($this->file)) {
+            if (is_dir($this->file)) {
+                throw new IOException(
+                    sprintf('Cannot write cache file "%s" as the location exists as directory.', realpath($this->file)),
+                    0,
+                    null,
+                    $this->file
+                );
+            }
+
+            if (!is_writable($this->file)) {
+                throw new IOException(
+                    sprintf('Cannot write to file "%s" as it is not writable.', realpath($this->file)),
+                    0,
+                    null,
+                    $this->file
+                );
+            }
+        } else {
+            @touch($this->file);
+            @chmod($this->file, 0666);
+        }
+
         $bytesWritten = @file_put_contents($this->file, $content, LOCK_EX);
 
         if (false === $bytesWritten) {
             $error = error_get_last();
 
             throw new IOException(
-                sprintf('Failed to write file "%s", "%s".', $this->file, $error ? $error['message'] : 'no reason available'),
+                sprintf('Failed to write file "%s", "%s".', $this->file, isset($error['message']) ? $error['message'] : 'no reason available'),
```

Comment:
you have added dedicated tests for 2 newly introduced `IOException` (big :+1: for that!),
is there a test for this `IOException` already?

Reply:
ok, there is.
I do believe that after solving my comments all of 3 test methods about `Symfony\Component\Filesystem\Exception\IOException` could be combined into 1 method with data provider

Reply:
I don't think so, the cases need different setups to trigger the exception. Merging those will give little gain and more work to maintain.